staging/src/k8s.io/client-go/tools/cache/thread_safe_store.go

```diff
@@ -185,7 +185,7 @@ func (c *threadSafeMap) ByIndex(indexName, indexKey string) ([]interface{}, erro
 
 	set := index[indexKey]
 	list := make([]interface{}, 0, set.Len())
-	for _, key := range set.List() {
```

Comment:
Looks like `sets.String.List()` would sort the keys, and this will be unsorted. Is that safe for all usages?

Reply:
It should be - noone should assume the fact that keys are sorted.
In fact we've done that for Index() function already some time ago:
https://github.com/kubernetes/kubernetes/pull/73529

Reply:
Great, thanks for explaining.